lib/ansible/modules/packaging/language/maven_artifact.py

```diff
@@ -425,7 +425,7 @@ def is_invalid_md5(self, file, remote_url):
                 parsed_url = urlparse(remote_url)
                 remote_md5 = self._local_md5(parsed_url.path)
             else:
-                remote_md5 = self._getContent(remote_url + '.md5', "Failed to retrieve MD5", False)
+                remote_md5 = self._getContent(remote_url + '.md5', "Failed to retrieve MD5", False).decode('utf-8', 'strict')
```

Comment:
Two things to change here:
* Use ansible.module_utils._text.to_text() instead of .decode().
* Catch the UnicodeError and give a better error message.  Maybe something like this:
``` python
from ansible.module_utils._text import to_native, to_text
[...]
    try:
        remote_md5 = to_text(self._getContent(remote_url + '.md5', "Failed to retrieve MD5", False), errors='strict')
    except UnicodeError as e:
        return "Cannot retrieve a valid md5 from %s: %s" % (remote_url, to_native(e))
```